### PR TITLE
Connect reports to db

### PR DESCRIPTION
**Pull request by david3214**
If you go to the reports page import and export Database should work as intended. Our default database file is database.csv

### CODE REVIEW

File: Database.py [hunk 1]
```diff
@@ -99,7 +100,7 @@ def importDB(self, importFPath: Path = Path("")) -> None:
                     if overwrite:
                         self.employeeList.remove(dupeList[0])
                         self.employeeList.append(emp)
-
+            self.exportDB(self.initFPath, adminInfo = True, showArchivedEmployees = True)
```
Comment (ComputationalPotato) [inline Database.py:103]: Don't force an auto save right after an import, the user might not want to depending on how the import goes.

File: Database.py [hunk 2]
```diff
@@ -114,6 +115,7 @@ def exportDB(self, exportPath: Path = Path(""), adminInfo: Boolean = False, show
         with open(exportPath, 'w', newline='') as csvfile:
             fieldnames = dir(EmployeeContainer)
             badnames = ['permissionList', 'getFName', 'getLName']
+            badnames = badnames if adminInfo else adminFields
```
Comment (ComputationalPotato) [inline Database.py:118]: I think should be if not adminInfo else badnames+adminFields

File: Address.py [hunk 3]
```diff
@@ -1,11 +1,10 @@
 class Address:
-    def __init__(self, address: str = "", apartment: str = "", city: str = "", state: str = "", country: str = "", zip: str = "") -> None:
-        self.address = address
-        self.apartment = apartment
-        self.city = city
-        self.state = state
-        self.country = country
-        self.zip = zip
+    def __init__(self, address: str = "", apartment: str = "", city: str = "", state: str = "", country: str = "", zip: str = "", **kwargs) -> None:
+        self.address = kwargs.get("Address", "")
```
Comment (ComputationalPotato) [inline Address.py:3]: Address class is not actually needed, it can be removed and its members move to be part of employee directly.(optional,low priority)

**Review by ComputationalPotato — APPROVED** [review]

File: Database.py (hunk 2 above)
Comment (david3214) [reply Database.py:118]: I can change that, right now I had that included in the admin fields, but that makes sense

File: Address.py (hunk 3 above)
Comment (david3214) [reply Address.py:3]: Yeah, I agree, I think we ended up doing it because the professor mentioned it. We can remove it at the end if we don't need it.

**Comment by ComputationalPotato** [comment]
this is not able to import files that it has exported because the keys are different for importing and exporting.

**Comment by david3214** [comment]
> this is not able to import files that it has exported because the keys are different for importing and exporting.

Can you expound further? it seems to be importing just fine for me. I added clarity on the hashed password just barely because it before wasn't taking it from either sheet, but it is working fine for me.

**Comment by david3214** [comment]
Export in general didn't work on the User Session branch until now because I forgot to add the user session into the export function, so maybe that's what you were noticing. The keys are working fine

**Comment by ComputationalPotato** [comment]
Export uses the names from EmployeeContainer as fieldnames while import
uses names from the professor's file or Employee

**Comment by david3214** [comment]
The import uses both. For the most part, they are the same, that's all there is. Have you tested the functionality or are you just stating what it looks like? because I tested it all and it works

**Comment by ComputationalPotato** [comment]
I tested on this current branch, where it is still __employee, and __employee works, while importing then exporting a file results in the loss of large amounts of data. kwargs is great at preventing errors, but errors are a good thing: they tell you that something went wrong.

**Comment by david3214** [comment]
Again, I don't know if you read clearly, the issues came on the other branch when the code was actually trying to access it. Not on this branch. This branch still uses __employee.

**Comment by ComputationalPotato** [comment]
I that we are waiting on this, that's why I approved the pr two days ago.

**Comment by ComputationalPotato** [comment]
also, social security number export key is 'SSNum' while import uses 'SSN'. this applies to several keys that are not in the professor's file.

**Comment by david3214** [comment]
You are right, I missed that one. I guess I just assumed that everything would be SSN

**Pull request merged**